streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -35,41 +35,34 @@
 import org.apache.kafka.streams.kstream.ValueJoiner;
 import org.apache.kafka.streams.kstream.ValueMapper;
 import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.test.ConsumerRecordFactory;
 import org.apache.kafka.test.KStreamTestDriver;
 import org.apache.kafka.test.MockAggregator;
 import org.apache.kafka.test.MockInitializer;
 import org.apache.kafka.test.MockMapper;
 import org.apache.kafka.test.MockProcessor;
 import org.apache.kafka.test.MockProcessorSupplier;
+import org.apache.kafka.test.StreamsTestUtils;
 import org.apache.kafka.test.TestUtils;
-import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 
-import java.io.File;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Properties;
 
 import static org.junit.Assert.assertEquals;
 
 public class KTableAggregateTest {
```

Comment:
A couple of tests in this class fail with `TopologyTestDriver`. My guess is that this is due to some problem with caching. Using `TopologyTestDriver`, regardless of the value I set for `StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG`, I get the same results out of `MockProcessor.processed` as if the cache was 0. In fact, setting the `KStreamTestDriver` cache to zero makes it yield the exact same results as `TopologyTestDriver`. I'll keep looking into it.  If you have any tip about what might be going wrong please let me know.

Reply:
Not entirely sure, but seems to be a bug in `TopologyTestDriver` related to `ThreadCache` -- `KStreamTestDriver` creates a cache and uses it as member variable and passes it into `store.init(...)` -- the new `TopologyTestDriver` also creates a cache and passes it into `GlobalProcessorContextImpl` and the created `StreamTask` -- maybe, internally the cache is not forwarded correctly such that it is not used when stores are initialized.

I would need to do a deeper analysis why the cache is not forwarded to the stores correctly. Hope the pointes help. If not sufficient, please let us know and we can dig deeper into it from our side.